autotest/gdrivers/mrf.py

```diff
@@ -233,6 +233,24 @@ def test_mrf_zen_test():
             gdal.Unlink(f)
 
 
+def test_mrf_in_tar():
+    import tarfile
+
+    gdal.Translate(
+        "plain.mrf", "data/byte.tif", format="MRF", creationOptions=["COMPRESS=DEFLATE"]
+    )
+    with tarfile.TarFile("plain.mrf.tar", "w", format=tarfile.GNU_FORMAT) as tar:
+        for ext in ("mrf", "idx", "pzp"):
+            tar.add("plain." + ext)
+    for ext in ("mrf", "idx", "pzp", "mrf.aux.xml"):
+        gdal.Unlink("plain." + ext)
+    ds = gdal.Open("plain.mrf.tar")
+    cs = ds.GetRasterBand(1).Checksum()
+    ds = None
+    assert cs == 4672
+    gdal.Unlink("plain.mrf.tar")
```

Comment:
would be better to write in tmp_path rather than in current directory. Something like:

```suggestion
def test_mrf_in_tar(tmp_path):
    import tarfile

    mrf_filename = str(tmp_path / "plain.mrf")   
    gdal.Translate(
        mrf_filename, "data/byte.tif", format="MRF", creationOptions=["COMPRESS=DEFLATE"]
    )
    
    mrf_tar_filename = str(tmp_path / "plain.mrf.tar")
    with tarfile.TarFile(mrf_tar_filename, "w", format=tarfile.GNU_FORMAT) as tar:
        for ext in ("mrf", "idx", "pzp"):
            tar.add(str(tmp_path / "plain." + ext))
    for ext in ("mrf", "idx", "pzp", "mrf.aux.xml"):
        gdal.Unlink(str(tmp_path / "plain." + ext))
    ds = gdal.Open(mrf_tar_filename)
    cs = ds.GetRasterBand(1).Checksum()
    ds = None
    assert cs == 4672
```

Reply:
The inside file can't have a path, to make it even less likely to trigger by mistake.
I did change the test to use the tmp_path, it is a better example of how to build such a tar in python